game_server/src/sc/server/gaming/GameRoom.java

```diff
@@ -519,8 +524,14 @@ public synchronized void onEvent(Client source, ProtocolMessage data)
 					"Game is already over, but got data: " + data.getClass());
 		}
 
-		this.game.onAction(resolvePlayer(source), data);
-	}
+    try {
+      this.game.onAction(resolvePlayer(source), data);
+    } catch (InvalidMoveException e) {
+      this.observerBroadcast(new RoomPacket(this.id, new ProtocolErrorMessage(e.getMove(), e.getMessage())));
+      throw new GameLogicException(e.getMessage());
```

Comment:
hier die ganze exception `e` und nicht nur die message uebergeben, sonst verliert man den trace

Reply:
wenn ich das richtig verstehe, wird die invalid move exception hier erst in eine game logic exception umgewandelt anstatt vorher schon in `RoundBasedGameInstance`. Kann das bei bestehenden clients probleme machen, weil sie eine game logic exception erwarten und eine invalid move exception bekommen?

Reply:
Das ganze Zeug passiert alles serverseitig, also erst, wenn der Server schon einen falschen Zug erkannt hat und das Spiel beenden will. Ich kann mir nicht vorstellen warum ein Client darauf warten sollte, weil der Server auch vorher davon nichts an die Clients gesendet hat. Die GameLogicException brauchen wir, damit das Spiel insgesamt beendet wird. Genau wie du erkannt hast wird die `InvalidMoveException` nur genutzt, damit wir im GameRoom an alle observer den letzten Zug mit der fehlermeldung senden können.